examples/oomkill.yaml

```
@@ -0,0 +1,48 @@
+programs:
+  # See:
+  # * https://github.com/iovisor/bcc/blob/master/tools/oomkill.py
+  # * https://github.com/iovisor/bcc/blob/master/tools/oomkill_example.txt
+  - name: oomkill
+    metrics:
+      counters:
+        - name: oom_kills
+          help: Count global and cgroup level OOMs
+          perf_map: events
+          labels:
+            - name: cgroup_path
+              size: 8
+              decoders:
+                - name: uint
+                - name: cgroup
+            - name: global_oom
+              size: 1
+              decoders:
+                - name: uint
+    kprobes:
+      oom_kill_process: count_ooms
+    code: |
+      #include <uapi/linux/ptrace.h>
+      #include <linux/oom.h>
+      #include <linux/memcontrol.h>
+
+      // we'll use "BPF_PERF_OUTPUT" map type here to avoid unbound cardinality
+      BPF_PERF_OUTPUT(events);
+
+      struct data_t {
+          u64 cgroup_id;
+          u8 global_oom;
+      };
+
+      void count_ooms(struct pt_regs *ctx, struct oom_control *oc, const char *message) {
+          struct data_t data = {};
+
+          struct mem_cgroup *mcg = oc->memcg;
+          if (!mcg) {
+              data.global_oom = 1;
+              events.perf_submit(ctx, &data, sizeof(data));
+              return;
+          }
+
+          data.cgroup_id = mcg->css.cgroup->kn->id.id;
```

We should probably aim for more recent kernels in examples, and since v5.5 it's just `id`:

* https://github.com/torvalds/linux/commit/67c0496e87d193b8356d2af49ab95e8a1b954b3c

I tested on v5.10:

```
$ sudo systemd-run --scope -p MemoryMax=2K ls
Killed
$ sudo systemd-run --scope -p MemoryMax=2K ls
Killed
```

```
$ curl -s http://localhost:9435/metrics | grep ^ebpf
ebpf_exporter_ebpf_programs{function="count_ooms",program="oomkill",tag="6539c1cc556ca54a"} 1
ebpf_exporter_enabled_programs{name="oomkill"} 1
ebpf_exporter_oom_kills{cgroup_path="/sys/fs/cgroup/system.slice/run-r32c99c50a92f4d9bb1db9bde34d94637.scope",global_oom="0"} 1
ebpf_exporter_oom_kills{cgroup_path="/sys/fs/cgroup/system.slice/run-r98efd5fe68254f9c827b88ba0b0b875e.scope",global_oom="0"} 1
```

LGTM otherwise.

```suggestion
          data.cgroup_id = mcg->css.cgroup->kn->id;
```